cpp/include/cudf/detail/iterator.cuh

```diff
@@ -427,7 +427,7 @@ struct scalar_optional_accessor : public scalar_value_accessor<Element> {
 
   __device__ inline value_type const operator()(size_type) const
   {
-    if (has_nulls && !super_t::dscalar.is_valid()) { return value_type{thrust::nullopt}; }
+    if (has_nulls && !super_t::dscalar.is_valid()) { return value_type{std::nullopt}; }
```

Comment:
In `src/replace/clamp.cu` we end up calling `(*x).value_or(...)` on the iterator value constructed here. But now this has changed from `thrust::optional::value_or` to `std::optional::value_or`, and nvcc complains that calling this `__host__` function in device code is invalid (https://github.com/rapidsai/cudf/actions/runs/8234015676/job/22514820431?pr=15091#step:7:1752)

Do we expect that to work?